roles/foreman/tasks/main.yaml

```diff
@@ -0,0 +1,95 @@
+  containers.podman.podman_secret:
+    state: present
+    name: foreman-database-url
+    data: "postgresql://foreman:{{ foreman_db_password }}@localhost/foreman"
+
+- name: Create katello config secret
+  containers.podman.podman_secret:
+    state: present
+    name: foreman-katello-yaml
+    data: "{{ lookup('ansible.builtin.template', 'katello.yaml.j2') }}"
+
+- name: Create the podman secret for Candlepin client certificate
+  containers.podman.podman_secret:
+    state: present
+    name: candlepin-client-cert
+    path: "{{ candlepin_client_certificate }}"
+
+- name: Create the podman secret for Candlepin client key
+  containers.podman.podman_secret:
+    state: present
+    name: candlepin-client-key
+    path: "{{ candlepin_client_key }}"
+
+- name: Deploy Foreman Container
+  containers.podman.podman_container:
+    name: "foreman"
+    image: "{{ foreman_image }}"
+    state: quadlet
+    ports: "{{ foreman_ports }}"
+    network: host
+    hostname: "{{ ansible_fqdn }}"
+    secrets:
+      - 'foreman-database-url,type=env,target=DATABASE_URL'
+      - 'foreman-katello-yaml,type=mount,target=/etc/foreman/plugins/katello.yaml'
+      - 'candlepin-ca-cert,type=mount,target=/etc/foreman/katello-default-ca.crt'
+      - 'candlepin-client-cert,type=mount,target=/etc/foreman/client_cert.pem'
+      - 'candlepin-client-key,type=mount,target=/etc/foreman/client_key.pem'
+    etc_hosts:
+      "quadlet.example.com": "127.0.0.1"
+    env:
+      SEED_ADMIN_PASSWORD: changeme
+
+- name: Deploy Dynflow Container
+  containers.podman.podman_container:
+    name: "dynflow-sidekiq-%i"
+    quadlet_filename: "dynflow-sidekiq@"
+    image: "{{ foreman_image }}"
+    state: quadlet
+    network: host
+    hostname: "{{ ansible_fqdn }}"
+    secrets:
+      - 'foreman-database-url,type=env,target=DATABASE_URL'
+      - 'foreman-katello-yaml,type=mount,target=/etc/foreman/plugins/katello.yaml'
+      - 'candlepin-ca-cert,type=mount,target=/etc/foreman/katello-default-ca.crt'
+      - 'candlepin-client-cert,type=mount,target=/etc/foreman/client_cert.pem'
+      - 'candlepin-client-key,type=mount,target=/etc/foreman/client_key.pem'
+    env:
+      DYNFLOW_SIDEKIQ_SCRIPT: "/usr/share/foreman/extras/dynflow-sidekiq.rb"
+      DYNFLOW_REDIS_URL: "redis://localhost:6379/6"
+      REDIS_PROVIDER: "DYNFLOW_REDIS_URL"
+    etc_hosts:
+      "quadlet.example.com": "127.0.0.1"
+    command: "/usr/libexec/foreman/sidekiq-selinux -e production -r /usr/share/foreman/extras/dynflow-sidekiq.rb -C /etc/foreman/dynflow/%i.yml"
+
+- name: Run daemon reload to make Quadlet create the service files
+  ansible.builtin.systemd:
+    daemon_reload: true
+
+- name: Start the Foreman Service
+  ansible.builtin.systemd:
+    name: foreman
+    enabled: true
+    state: started
+
+- name: Start the Dynflow Services
+  ansible.builtin.systemd:
+    name: "dynflow-sidekiq@{{ item }}"
+    enabled: true
+    state: started
+  loop:
+    - orchestrator
+    - worker
```

Comment:
this is missing the `hosts_queue` worker, because it's not in the default config and I was too lazy to create a config yet

needed for Katello

Reply:
#18